internal/cmd/gocdk/_static/demo/pubsub/demo_pubsub.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"context"
+	"html/template"
+	"net/http"
+	"os"
+	"time"
+
+	"gocloud.dev/pubsub"
+	_ "gocloud.dev/pubsub/mempubsub"
+)
+
+// TODO(rvangent): This file is user-visible, add many comments explaining
+// how it works.
+
+// How long to wait for a message before giving up.
+const waitForMessage = 500 * time.Millisecond
+
+func init() {
+	http.HandleFunc("/demo/pubsub/", pubsubSendHandler)
+	http.HandleFunc("/demo/pubsub/send", pubsubSendHandler)
+	http.HandleFunc("/demo/pubsub/receive", pubsubReceiveHandler)
+}
+
+var topicURL string
+var topic *pubsub.Topic
+var topicErr error
+
+var subscriptionURL string
+var subscription *pubsub.Subscription
+var subscriptionErr error
+
+func init() {
+	topicURL = os.Getenv("PUBSUB_TOPIC_URL")
```

Comment:
This is not specific to this PR, but I do wonder if we want to preface/namespace our demo env vars to make them easier for the user to clean up? or should that be another cli operation? (remove demo)

Reply:
IIUC, the environment variables will be set based on the `outputs.tf`, so they won't have to clean them up. We might still want to namespace them though. I suspect this will come up with we start setting them based on `outputs.tf`.